--- ntfs-3g-2009.4.4-origin/src/Makefile.am	2013-01-13 16:22:27.962435490 +0800
+++ ntfs-3g-2009.4.4/src/Makefile.am	2013-01-13 16:23:26.310432889 +0800
@@ -41,6 +41,7 @@
 install-exec-local:	install-rootbinPROGRAMS
 	$(MKDIR_P) "$(DESTDIR)/sbin"
 	$(LN_S) -f "$(rootbindir)/ntfs-3g" "$(DESTDIR)/sbin/mount.ntfs-3g"
+	$(LN_S) -f "$(rootbindir)/ntfs-3g" "$(DESTDIR)/sbin/mount.ntfs"
 endif
 
 install-data-local:	install-man8
@@ -50,4 +51,5 @@
 	$(RM) -f "$(DESTDIR)$(man8dir)/mount.ntfs-3g.8"
 if ENABLE_MOUNT_HELPER
 	$(RM) -f "$(DESTDIR)/sbin/mount.ntfs-3g"
+	$(RM) -f "$(DESTDIR)/sbin/mount.ntfs"
 endif
